<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>CytoGLMM Workflow • CytoGLMM</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="CytoGLMM Workflow">
<meta property="og:description" content="CytoGLMM">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">CytoGLMM</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.3.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/CytoGLMM.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/pregnancy_dataset.html">CytoGLMM Workflow for Pregnancy Dataset</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/ChristofSeiler/CytoGLMM/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="CytoGLMM_files/header-attrs-2.11/header-attrs.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>CytoGLMM Workflow</h1>
                        <h4 class="author">Christof Seiler</h4>
            <address class="author_afil">
      Department of Statistics, Stanford University &amp; Department of Data Science and Knowledge Engineering, Maastricht University<br><h4 class="date">2021-10-30</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/ChristofSeiler/CytoGLMM/blob/pregnancy/vignettes/CytoGLMM.Rmd"><code>vignettes/CytoGLMM.Rmd</code></a></small>
      <div class="hidden name"><code>CytoGLMM.Rmd</code></div>

    </address>
</div>

    
    
<div id="introduction" class="section level1">
<h1 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h1>
<p>Flow and mass cytometry are important modern immunology tools for measuring expression levels of multiple proteins on single cells. The goal is to better understand the mechanisms of responses on a single cell basis by studying differential expression of proteins. Most current data analysis tools compare expressions across many computationally discovered cell types. Our goal is to focus on just one cell type. Differential analysis of marker expressions can be difficult due to marker correlations and inter-subject heterogeneity, particularly for studies of human immunology. We address these challenges with two multiple regression strategies: A bootstrapped generalized linear model (GLM) and a generalized linear mixed model (GLMM). Here, we illustrate the <code>CytoGLMM</code> <em>R</em> package and workflow for simulated mass cytometry data.</p>
</div>
<div id="prepare-simulated-data" class="section level1">
<h1 class="hasAnchor">
<a href="#prepare-simulated-data" class="anchor"></a>Prepare Simulated Data</h1>
<p>We construct our simulated datasets by sampling from a Poisson GLM. We confirmed—with predictive posterior checks—that Poisson GLMs with mixed effects provide a good fit to mass cytometry data <span class="citation">(Seiler et al. 2019)</span>. We consider one underlying data generating mechanisms described by a hierarchical model for the <span class="math inline">\(i\)</span>th cell and <span class="math inline">\(j\)</span>th donor:</p>
<p><span class="math display">\[
\begin{aligned}
\boldsymbol{X}_{ij} &amp; \sim \text{Poisson}(\boldsymbol{\lambda}_{ij}) \\
\log(\boldsymbol{\lambda}_{ij}) &amp; = \boldsymbol{B}_{ij} + \boldsymbol{U}_j \\
\boldsymbol{B}_{ij} &amp; \sim
\begin{cases} 
\text{Normal}(\boldsymbol{\delta}^{(0)}, \boldsymbol{\Sigma}_B) &amp; \text{if } Y_{ij} = 0, \text{ cell unstimulated} \\
\text{Normal}(\boldsymbol{\delta}^{(1)}, \boldsymbol{\Sigma}_B) &amp; \text{if } Y_{ij} = 1, \text{ cell stimulated}
\end{cases} \\
\boldsymbol{U}_j &amp; \sim \text{Normal}(\boldsymbol{0}, \boldsymbol{\Sigma}_U).
\end{aligned}
\]</span></p>
<p>The following graphic shows a representation of the hierarchical model.</p>
<p><img src="graphical_plmm.svg"></p>
<p>The stimulus activates proteins and induces a difference in marker expression. We define the effect size to be the difference between expected expression levels of stimulated versus unstimulated cells on the <span class="math inline">\(\log\)</span>-scale. All markers that belong to the active set , have a non-zero effect size, whereas, all markers that are not, have a zero effect size:</p>
<p><span class="math display">\[
\begin{cases}
\delta^{(1)}_p - \delta^{(0)}_p &gt; 0 &amp; \text{if protein } p \text{ is in activation set } p \in C \\
\delta^{(1)}_{p'} - \delta^{(0)}_{p'} = 0 &amp; \text{if protein } p' \text{ is not in activation set } p' \notin C.
\end{cases}
\]</span></p>
<p>Both covariance matrices have an autoregressive structure,</p>
<p><span class="math display">\[
\begin{aligned}
\Omega_{rs} &amp; = \rho^{|r-s|} \\ 
\boldsymbol{\Sigma} &amp; = \operatorname{diag}(\boldsymbol{\sigma}) \, \boldsymbol{\Omega} \, \operatorname{diag}(\boldsymbol{\sigma}),
\end{aligned}
\]</span></p>
<p>where <span class="math inline">\(\Omega_{rs}\)</span> is the <span class="math inline">\(r\)</span>th row and <span class="math inline">\(s\)</span>th column of the correlation matrix <span class="math inline">\(\boldsymbol{\Omega}\)</span>. We regulate two separate correlation parameters: a cell-level <span class="math inline">\(\rho_B\)</span> and a donor-level <span class="math inline">\(\rho_U\)</span> coefficient. Non-zero <span class="math inline">\(\rho_B\)</span> or <span class="math inline">\(\rho_U\)</span> induce a correlation between condition and marker expression even for markers with a zero effect size.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="st"><a href="https://christofseiler.github.io/CytoGLMM">"CytoGLMM"</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="st"><a href="https://magrittr.tidyverse.org">"magrittr"</a></span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/Random.html">set.seed</a></span><span class="op">(</span><span class="fl">23</span><span class="op">)</span>
<span class="va">df</span> <span class="op">=</span> <span class="fu"><a href="../reference/generate_data.html">generate_data</a></span><span class="op">(</span><span class="op">)</span>
<span class="va">df</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span>,<span class="fl">1</span><span class="op">:</span><span class="fl">5</span><span class="op">]</span></code></pre></div>
<pre><code>## # A tibble: 5 × 5
##   donor condition   m01   m02   m03
##   &lt;int&gt; &lt;fct&gt;     &lt;int&gt; &lt;int&gt; &lt;int&gt;
## 1     1 treatment     2   127   116
## 2     1 treatment     0    96    54
## 3     1 treatment     1    14    52
## 4     1 treatment     6    20    34
## 5     1 treatment    18    30    15</code></pre>
<p>We define the marker names that we will focus on in our analysis by extracting them from the simulated data frame.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">protein_names</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">df</span><span class="op">)</span><span class="op">[</span><span class="fl">3</span><span class="op">:</span><span class="fl">12</span><span class="op">]</span></code></pre></div>
<p>We recommend that marker expressions be corrected for batch effects <span class="citation">(Nowicka et al. 2017; Chevrier et al. 2018; Schuyler et al. 2019; Van Gassen et al. 2020; Trussart et al. 2020)</span> and transformed using variance stabilizing transformations to account for heteroskedasticity, for instance with an inverse hyperbolic sine transformation with the cofactor set to 150 for flow cytometry, and 5 for mass cytometry <span class="citation">(Bendall et al. 2011)</span>. This transformation assumes a two-component model for the measurement error <span class="citation">(Rocke and Lorenzato 1995; Huber et al. 2003)</span> where small counts are less noisy than large counts. Intuitively, this corresponds to a noise model with additive and multiplicative noise depending on the magnitude of the marker expression; see <span class="citation">(Holmes and Huber 2019)</span> for details.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">df</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/compound.html">%&lt;&gt;%</a></span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate_all.html">mutate_at</a></span><span class="op">(</span><span class="va">protein_names</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/r/base/Hyperbolic.html">asinh</a></span><span class="op">(</span><span class="va">x</span><span class="op">/</span><span class="fl">5</span><span class="op">)</span><span class="op">)</span></code></pre></div>
</div>
<div id="glm" class="section level1">
<h1 class="hasAnchor">
<a href="#glm" class="anchor"></a>GLM</h1>
<p>The goal of the <code><a href="../reference/cytoglm.html">CytoGLMM::cytoglm</a></code> function is to find protein expression patterns that are associated with the condition of interest, such as a response to a stimulus. We set up the GLM to predict the experimental condition (<code>condition</code>) from protein marker expressions (<code>protein_names</code>), thus our experimental conditions are response variables and marker expressions are explanatory variables.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">glm_fit</span> <span class="op">=</span> <span class="fu">CytoGLMM</span><span class="fu">::</span><span class="fu"><a href="../reference/cytoglm.html">cytoglm</a></span><span class="op">(</span><span class="va">df</span>,
                            protein_names <span class="op">=</span> <span class="va">protein_names</span>,
                            condition <span class="op">=</span> <span class="st">"condition"</span>,
                            group <span class="op">=</span> <span class="st">"donor"</span>,
                            num_cores <span class="op">=</span> <span class="fl">1</span>,
                            num_boot <span class="op">=</span> <span class="fl">1000</span><span class="op">)</span>
<span class="va">glm_fit</span></code></pre></div>
<pre><code>## 
## #######################
## ## paired analysis ####
## #######################
## 
## number of bootstrap samples: 1000 
## 
## number of cells per group and condition:   
##     control treatment
##   1    1000      1000
##   2    1000      1000
##   3    1000      1000
##   4    1000      1000
##   5    1000      1000
##   6    1000      1000
##   7    1000      1000
##   8    1000      1000
## 
## proteins included in the analysis:
##  m01 m02 m03 m04 m05 m06 m07 m08 m09 m10 
## 
## condition compared: condition 
## grouping variable: donor</code></pre>
<p>We plot the maximum likelihood estimates with 95% confidence intervals for the fixed effects <span class="math inline">\(\boldsymbol{\beta}\)</span>. The estimates are on the <span class="math inline">\(\log\)</span>-odds scale. We see that markers m1, m2, and m3 are strong predictors of the treatment. This means that one unit increase in the transformed marker expression makes it more likely to be a cell from the treatment group, while holding the other markers constant.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html">plot</a></span><span class="op">(</span><span class="va">glm_fit</span><span class="op">)</span></code></pre></div>
<p><img src="CytoGLMM_files/figure-html/glm_plot-1.png" width="700.0000032"></p>
<p>The <code>summary</code> function returns a table about the model fit with unadjusted and Benjamini-Hochberg (BH) adjusted <span class="math inline">\(p\)</span>-values.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/summary.html">summary</a></span><span class="op">(</span><span class="va">glm_fit</span><span class="op">)</span></code></pre></div>
<pre><code>## # A tibble: 10 × 3
##    protein_name pvalues_unadj pvalues_adj
##    &lt;chr&gt;                &lt;dbl&gt;       &lt;dbl&gt;
##  1 m03                  0.002       0.02 
##  2 m02                  0.004       0.02 
##  3 m01                  0.018       0.06 
##  4 m06                  0.356       0.867
##  5 m04                  0.496       0.867
##  6 m08                  0.6         0.867
##  7 m05                  0.684       0.867
##  8 m09                  0.694       0.867
##  9 m10                  0.868       0.962
## 10 m07                  0.962       0.962</code></pre>
<p>We can extract the proteins below an False Discovery Rate (FDR) of 0.05 from the <span class="math inline">\(p\)</span>-value table by filtering the table.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/summary.html">summary</a></span><span class="op">(</span><span class="va">glm_fit</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span><span class="op">(</span><span class="va">pvalues_adj</span> <span class="op">&lt;</span> <span class="fl">0.05</span><span class="op">)</span></code></pre></div>
<pre><code>## # A tibble: 2 × 3
##   protein_name pvalues_unadj pvalues_adj
##   &lt;chr&gt;                &lt;dbl&gt;       &lt;dbl&gt;
## 1 m03                  0.002        0.02
## 2 m02                  0.004        0.02</code></pre>
<p>In this simulated dataset, the markers m2 and m3 are below an FDR of 0.05.</p>
</div>
<div id="glmm" class="section level1">
<h1 class="hasAnchor">
<a href="#glmm" class="anchor"></a>GLMM</h1>
<p>In the <code><a href="../reference/cytoglmm.html">CytoGLMM::cytoglmm</a></code> function, we make additional modeling assumptions by adding a random effect term in the standard logistic regression model to account for the subject effect (<code>group</code>). In paired experimental design—when the same donor provides two samples, one for each condition—<code><a href="../reference/cytoglmm.html">CytoGLMM::cytoglmm</a></code> is statistically more powerful.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">glmm_fit</span> <span class="op">=</span> <span class="fu">CytoGLMM</span><span class="fu">::</span><span class="fu"><a href="../reference/cytoglmm.html">cytoglmm</a></span><span class="op">(</span><span class="va">df</span>,
                              protein_names <span class="op">=</span> <span class="va">protein_names</span>,
                              condition <span class="op">=</span> <span class="st">"condition"</span>,
                              group <span class="op">=</span> <span class="st">"donor"</span>,
                              num_cores <span class="op">=</span> <span class="fl">1</span><span class="op">)</span>
<span class="va">glmm_fit</span></code></pre></div>
<pre><code>## number of cells per group and condition:   
##     control treatment
##   1    1000      1000
##   2    1000      1000
##   3    1000      1000
##   4    1000      1000
##   5    1000      1000
##   6    1000      1000
##   7    1000      1000
##   8    1000      1000
## 
## proteins included in the analysis:
##  m01 m02 m03 m04 m05 m06 m07 m08 m09 m10 
## 
## condition compared: condition 
## grouping variable: donor</code></pre>
<p>We plot the method of moments estimates with 95% confidence intervals for the fixed effects <span class="math inline">\(\boldsymbol{\beta}\)</span>.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html">plot</a></span><span class="op">(</span><span class="va">glmm_fit</span><span class="op">)</span></code></pre></div>
<p><img src="CytoGLMM_files/figure-html/glmm_plot-1.png" width="700.0000032"></p>
<p>The <code>summary</code> function returns a table about the model fit with unadjusted and BH adjusted <span class="math inline">\(p\)</span>-values.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/summary.html">summary</a></span><span class="op">(</span><span class="va">glmm_fit</span><span class="op">)</span></code></pre></div>
<pre><code>## # A tibble: 10 × 3
##    protein_name pvalues_unadj pvalues_adj
##    &lt;chr&gt;                &lt;dbl&gt;       &lt;dbl&gt;
##  1 m03              0.0000682    0.000682
##  2 m02              0.000163     0.000813
##  3 m01              0.000681     0.00227 
##  4 m06              0.155        0.388   
##  5 m05              0.371        0.646   
##  6 m08              0.388        0.646   
##  7 m04              0.512        0.731   
##  8 m09              0.859        0.954   
##  9 m10              0.874        0.954   
## 10 m07              0.954        0.954</code></pre>
<p>We can extract the proteins below an FDR of 0.05 from the <span class="math inline">\(p\)</span>-value table by filtering the table.</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/summary.html">summary</a></span><span class="op">(</span><span class="va">glmm_fit</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span><span class="op">(</span><span class="va">pvalues_adj</span> <span class="op">&lt;</span> <span class="fl">0.05</span><span class="op">)</span></code></pre></div>
<pre><code>## # A tibble: 3 × 3
##   protein_name pvalues_unadj pvalues_adj
##   &lt;chr&gt;                &lt;dbl&gt;       &lt;dbl&gt;
## 1 m03              0.0000682    0.000682
## 2 m02              0.000163     0.000813
## 3 m01              0.000681     0.00227</code></pre>
<p>In this simulated dataset, the markers m1, m2 and m3 are below an FDR of 0.05. This illustrates the improved power for paired samples as <code><a href="../reference/cytoglmm.html">CytoGLMM::cytoglmm</a></code> correctly detects all three differentially expressed proteins. We performed extensive power simulations and comparisons to other methods in <span class="citation">Seiler et al. (2021)</span>.</p>
</div>
<div id="session-info" class="section level1 unnumbered">
<h1 class="hasAnchor">
<a href="#session-info" class="anchor"></a>Session Info</h1>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code>## R version 4.1.0 (2021-05-18)
## Platform: x86_64-apple-darwin17.0 (64-bit)
## Running under: macOS Big Sur 10.16
## 
## Matrix products: default
## BLAS:   /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRblas.dylib
## LAPACK: /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRlapack.dylib
## 
## locale:
## [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8
## 
## attached base packages:
## [1] stats     graphics  grDevices utils     datasets  methods   base     
## 
## other attached packages:
## [1] magrittr_2.0.1   CytoGLMM_1.3.0   BiocStyle_2.22.0
## 
## loaded via a namespace (and not attached):
##   [1] bigmemory_4.5.36     bigmemory.sri_0.1.3  minqa_1.2.4         
##   [4] colorspace_2.0-2     ellipsis_0.3.2       class_7.3-19        
##   [7] modeltools_0.2-23    rprojroot_2.0.2      fs_1.5.0            
##  [10] rstudioapi_0.13      listenv_0.8.0        farver_2.1.0        
##  [13] ggrepel_0.9.1        flexmix_2.3-17       prodlim_2019.11.13  
##  [16] fansi_0.5.0          lubridate_1.8.0      codetools_0.2-18    
##  [19] splines_4.1.0        logging_0.10-108     doParallel_1.0.16   
##  [22] cachem_1.0.6         knitr_1.36           speedglm_0.3-3      
##  [25] nloptr_1.2.2.2       pROC_1.18.0          caret_6.0-90        
##  [28] pheatmap_1.0.12      BiocManager_1.30.16  compiler_4.1.0      
##  [31] assertthat_0.2.1     Matrix_1.3-4         fastmap_1.1.0       
##  [34] strucchange_1.5-2    cli_3.1.0            htmltools_0.5.2     
##  [37] tools_4.1.0          gtable_0.3.0         glue_1.4.2          
##  [40] reshape2_1.4.4       dplyr_1.0.7          Rcpp_1.0.7          
##  [43] jquerylib_0.1.4      pkgdown_1.6.1        vctrs_0.3.8         
##  [46] nlme_3.1-153         iterators_1.0.13     mbest_0.6           
##  [49] timeDate_3043.102    gower_0.2.2          xfun_0.27           
##  [52] stringr_1.4.0        globals_0.14.0       lme4_1.1-27.1       
##  [55] lifecycle_1.0.1      future_1.22.1        factoextra_1.0.7    
##  [58] MASS_7.3-54          zoo_1.8-9            scales_1.1.1        
##  [61] ipred_0.9-12         ragg_1.1.3           parallel_4.1.0      
##  [64] sandwich_3.0-1       RColorBrewer_1.1-2   yaml_2.2.1          
##  [67] memoise_2.0.0        ggplot2_3.3.5        rpart_4.1-15        
##  [70] stringi_1.7.5        highr_0.9            desc_1.4.0          
##  [73] foreach_1.5.1        boot_1.3-28          BiocParallel_1.28.0 
##  [76] lava_1.6.10          rlang_0.4.12         pkgconfig_2.0.3     
##  [79] systemfonts_1.0.3    evaluate_0.14        lattice_0.20-45     
##  [82] purrr_0.3.4          recipes_0.1.17       labeling_0.4.2      
##  [85] cowplot_1.1.1        tidyselect_1.1.1     parallelly_1.28.1   
##  [88] plyr_1.8.6           bookdown_0.24        R6_2.5.1            
##  [91] generics_0.1.1       DBI_1.1.1            pillar_1.6.4        
##  [94] withr_2.4.2          survival_3.2-13      abind_1.4-5         
##  [97] nnet_7.3-16          tibble_3.1.5         future.apply_1.8.1  
## [100] crayon_1.4.2         utf8_1.2.2           rmarkdown_2.11      
## [103] grid_4.1.0           data.table_1.14.2    ModelMetrics_1.2.2.2
## [106] digest_0.6.28        tidyr_1.1.4          textshaping_0.3.6   
## [109] stats4_4.1.0         munsell_0.5.0</code></pre>
</div>
<div id="references" class="section level1 unnumbered">
<h1 class="hasAnchor">
<a href="#references" class="anchor"></a>References</h1>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-bendall2011single" class="csl-entry">
Bendall, Sean C, Erin F Simonds, Peng Qiu, D Amir El-ad, Peter O Krutzik, Rachel Finck, Robert V Bruggner, et al. 2011. <span>“Single-Cell Mass Cytometry of Differential Immune and Drug Responses Across a Human Hematopoietic Continuum.”</span> <em>Science</em> 332 (6030): 687–96.
</div>
<div id="ref-chevrier2018compensation" class="csl-entry">
Chevrier, Stéphane, Helena L Crowell, Vito RT Zanotelli, Stefanie Engler, Mark D Robinson, and Bernd Bodenmiller. 2018. <span>“Compensation of Signal Spillover in Suspension and Imaging Mass Cytometry.”</span> <em>Cell Systems</em> 6 (5): 612–20.
</div>
<div id="ref-holmes2019modern" class="csl-entry">
Holmes, Susan, and Wolfgang Huber. 2019. <em>Modern Statistics for Modern Biology</em>. Cambridge University Press.
</div>
<div id="ref-huber2003parameter" class="csl-entry">
Huber, Wolfgang, Anja von Heydebreck, Holger Sültmann, Annemarie Poustka, and Martin Vingron. 2003. <span>“Parameter Estimation for the Calibration and Variance Stabilization of Microarray Data.”</span> <em>Statistical Applications in Genetics and Molecular Biology</em> 2 (1).
</div>
<div id="ref-nowicka2017cytof" class="csl-entry">
Nowicka, M, C Krieg, LM Weber, FJ Hartmann, S Guglietta, B Becher, MP Levesque, and MD Robinson. 2017. <span>“CyTOF Workflow: <span>D</span>ifferential Discovery in High-Throughput High-Dimensional Cytometry Datasets [Version 2; Referees: 2 Approved].”</span> <em>F1000Research</em> 6 (748).
</div>
<div id="ref-rocke1995two" class="csl-entry">
Rocke, David M, and Stefan Lorenzato. 1995. <span>“A Two-Component Model for Measurement Error in Analytical Chemistry.”</span> <em>Technometrics</em> 37 (2): 176–84.
</div>
<div id="ref-schuyler2019minimizing" class="csl-entry">
Schuyler, Ronald P., Conner Jackson, Josselyn E. Garcia-Perez, Ryan M. Baxter, Sidney Ogolla, Rosemary Rochford, Debashis Ghosh, Pratyaydipta Rudra, and Elena W. Y. Hsieh. 2019. <span>“Minimizing Batch Effects in Mass Cytometry Data.”</span> <em>Frontiers in Immunology</em> 10: 2367.
</div>
<div id="ref-seiler2021cytoglmm" class="csl-entry">
Seiler, Christof, Anne-Maud Ferreira, Lisa M Kronstad, Laura J Simpson, Mathieu Le Gars, Elena Vendrame, Catherine A Blish, and Susan Holmes. 2021. <span>“CytoGLMM: Conditional Differential Analysis for Flow and Mass Cytometry Experiments.”</span> <em>BMC Bioinformatics</em> 22 (137): 1–14.
</div>
<div id="ref-seiler2019uncertainty" class="csl-entry">
Seiler, Christof, Lisa M Kronstad, Laura J Simpson, Mathieu Le Gars, Elena Vendrame, Catherine A Blish, and Susan Holmes. 2019. <span>“Uncertainty Quantification in Multivariate Mixed Models for Mass Cytometry Data.”</span> <em>arXiv:1903.07976</em>.
</div>
<div id="ref-trussart2020removing" class="csl-entry">
Trussart, Marie, Charis E Teh, Tania Tan, Lawrence Leong, Daniel HD Gray, and Terence P Speed. 2020. <span>“Removing Unwanted Variation with <span>CytofRUV</span> to Integrate Multiple <span>CyTOF</span> Datasets.”</span> <em>eLife</em> 9 (September): e59630.
</div>
<div id="ref-van2020cytonorm" class="csl-entry">
Van Gassen, Sofie, Brice Gaudilliere, Martin S Angst, Yvan Saeys, and Nima Aghaeepour. 2020. <span>“<span>CytoNorm</span>: A Normalization Algorithm for Cytometry Data.”</span> <em>Cytometry Part A</em> 97 (3): 268–78.
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Christof Seiler.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
